.github/workflows/main.yml

```diff
@@ -0,0 +1,23 @@
+on: [pull_request]
+jobs:
+  build:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: 'meta-rust'
+      - name: Fetch poky
+        run: |
+          mv poky/build/sstate-cache . || true
+          rm -rf poky meta-openembedded
+          git clone -b gatesgarth --single-branch git://git.yoctoproject.org/poky
+          git clone -b gatesgarth --single-branch git://git.openembedded.org/meta-openembedded
+      - name: Run bitbake
+        run: |
+          cd poky
+          . oe-init-build-env
+          mv ../../sstate-cache . || true
+          sed -i -e "\$i$(readlink -f ../../meta-openembedded/meta-oe) \\\\" conf/bblayers.conf
+          sed -i -e "\$i$(readlink -f ../../meta-rust) \\\\" conf/bblayers.conf
```

Comment:
I think there's a `bitbake-layers` tool that we can use for this.

Reply:
I use GH actions for CI for YoeDistro regularly and also for other layers I maintain see

see how its done for meta-clang, where it checks out distro layers and replaces meta-clang with the pull branch smartly.
https://github.com/kraj/meta-clang/blob/master/.github/workflows/yoe.yml

yoe CI
https://github.com/YoeDistro/yoe-distro/blob/master/.github/workflows/yoe.yml

maybe it will be helpful